src/main/java/database/service/service/DatabaseService.java

```diff
@@ -0,0 +1,36 @@
+package database.service.service;
+
+import code.practice.exceptions.DatabaseDoesNotExistException;
+import code.practice.exceptions.ProvidedIdDoesNotExistException;
+import code.practice.exceptions.TableIsNotCompatibleWithEntityException;
+import database.service.entity.BaseEntity;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.IOException;
+import java.util.Set;
+
+public interface DatabaseService {
```

Comment:
Please remove all throws statements since we can implement this interface for different types of a databases and not all of them can produce IOException , DatabaseDoesNotExistException and others.